src/tree-view/index.tsx

```diff
@@ -0,0 +1,29 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React from 'react';
+
+import useBaseComponent from '../internal/hooks/use-base-component';
+import { applyDisplayName } from '../internal/utils/apply-display-name';
+import { TreeViewProps } from './interfaces';
+
+export { TreeViewProps };
+
+/**
+ * @awsuiSystem core
```

Comment:
With this tag, the component will not be exposed in any public artifacts, even after merging to mainline

Reply:
How is that ensured?

Reply:
Updated internal part now, all works as intended